### ISSUE TITLE
Warn if user calls `.to_string()` multiple times on variable.

### ISSUE THREAD

**Issue by Tanguille** [rust-lang/rust-clippy#12152]
### What it does

Warn that `.to_string()` is only required once. Even with pedantic clippy this doesn't give any warning.

### Advantage

It can happen to anyone if you process a lot of code that you accidently call `.to_string()` more then once on a variable. It takes up useless characters in your file and looks sloppy. This looks like something clippy could fix automatically with the `--fix` argument.

### Drawbacks

I can't think of any since I don't know any situation where it would make sense to call `.to_string()` twice in a variable.

### Example

```rust
 let uri = req.uri().path().to_string().to_string();
```

Could be written as:

```rust
 let uri = req.uri().path().to_string();
```

**Comment by y21** [comment]
This lint exists: https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_clone
With `#![warn(clippy::redundant_clone)]`:

```
warning: redundant clone
 --> src/main.rs:6:9
  |
6 |         .to_string()
  |         ^^^^^^^^^^^^ help: remove this
  |
note: this value is dropped without further use
 --> src/main.rs:4:5
  |
4 | /     x
5 | |         .to_string()
6 | |         .to_string()
  | |________^
```

The lint is currently opt-in because it has way too many false positive, see: https://github.com/rust-lang/rust-clippy/issues?q=is%3Aissue+redundant_clone


The PR #11364 rewrites it
